include/exec/any_sender_of.hpp

```diff
@@ -612,16 +614,42 @@ namespace exec {
     auto __tag_type(_Tag (*)(_As...) noexcept) -> _Tag;
 
     template <class _Query>
-    concept __is_stop_token_query = requires {
-      { __tag_type(static_cast<_Query>(nullptr)) } -> same_as<get_stop_token_t>;
-    };
+    using __tag_type_t = decltype(__tag_type(static_cast<_Query>(nullptr)));
+
+    template <class _Query>
+    concept __is_stop_token_query = __same_as<__tag_type_t<_Query>, get_stop_token_t>;
 
     template <class _Query>
     concept __is_not_stop_token_query = !__is_stop_token_query<_Query>;
 
     template <class _Query>
     using __is_not_stop_token_query_v = __mbool<__is_not_stop_token_query<_Query>>;
 
+    auto __test_never_stop_token(get_stop_token_t (*)(never_stop_token (*)() noexcept))
+      -> __mbool<true>;
+
+    template <class _Tag, class _Ret, class... _As>
+    auto __test_never_stop_token(_Tag (*)(_Ret (*)(_As...) noexcept)) -> __mbool<false>;
+
+    template <class _Tag, class _Ret, class... _As>
+    auto __test_never_stop_token(_Tag (*)(_Ret (*)(_As...))) -> __mbool<false>;
+
+    template <class _Query>
+    using __is_never_stop_token_query_v = decltype(__test_never_stop_token(
```

Comment:
the `_v` suffix is typically used for variable templates. this is an alias template, so `_t` would be more appropriate.

Reply:
yap, fixed.